executor/builder.go

```diff
@@ -1063,10 +1111,17 @@ func (b *executorBuilder) buildIndexLookUpReader(v *plan.PhysicalIndexLookUpRead
 	}
 	is := v.IndexPlans[0].(*plan.PhysicalIndexScan)
 	table, _ := b.is.TableByID(is.Table.ID)
+	var handleCol *expression.Column
+	if v.NeedColHandle {
+		handleCol = v.Schema().TblID2handle[is.Table.ID][0]
+	}
 
 	for i := range v.Schema().Columns {
 		tableReq.OutputOffsets = append(tableReq.OutputOffsets, uint32(i))
 	}
+	if v.NeedColHandle && handleCol.ID == -1 {
```

Comment:
If `handleCol.ID = -1`, `v.NeedColHandle` must be ture. This check can also be moved to 1120.